lib/fw.c

```diff
@@ -803,31 +863,32 @@ int switchtec_fw_read_active_map_footer(struct switchtec_dev *dev,
 	if (map0_update_index < map1_update_index)
 		active_map_part_start = SWITCHTEC_FLASH_MAP1_PART_START;
 
-
-	return switchtec_fw_read_footer(dev, active_map_part_start,
-					SWITCHTEC_FLASH_PART_LEN, ftr, version,
-					version_len);
+	return switchtec_fw_read_metadata(dev, active_map_part_start,
+				      SWITCHTEC_FLASH_PART_LEN, ftr, version,
+				      version_len);
 }
 
 /**
  * @brief Write the header for a Switchtec firmware image file
  * @param[in]  fd	File descriptor for image file to write
- * @param[in]  ftr	Footer information to include in the header
+ * @param[in]  ftr	switchtec_fwdl_meta information to include in the header
  * @param[in]  type	File type to record in the header
  * @return 0 on success, error code on failure
  */
-int switchtec_fw_img_write_hdr(int fd, struct switchtec_fw_footer *ftr,
+int switchtec_fw_img_write_hdr(int fd, struct switchtec_fwdl_meta *ftr,
 			       enum switchtec_fw_image_type type)
 {
+	enum switchtec_fw_image_type imgtype;
 	struct fw_image_header hdr = {};
 
-	memcpy(hdr.magic, ftr->magic, sizeof(hdr.magic));
-	hdr.image_len = ftr->image_len;
-	hdr.type = type;
-	hdr.load_addr = ftr->load_addr;
-	hdr.version = ftr->version;
-	hdr.header_crc = ftr->header_crc;
-	hdr.image_crc = ftr->image_crc;
+	imgtype = type;
+	memcpy(hdr.vendor_id, ftr->vendor_id, sizeof(hdr.vendor_id));
```

Comment:
why not keep original code?

Reply:
yes, modify to the same of saratoga code style.